core/interfaces/i_toolbox.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The interface for a toolbox.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+'use strict';
+
+goog.provide('Blockly.IToolbox');
+
+goog.require('Blockly.IRegistrable');
+
+
+/**
+ * @extends {Blockly.IRegistrable}
+ * @interface
+ */
+Blockly.IToolbox = function() {};
+
+/**
+ * Initializes the toolbox.
+ */
+Blockly.IToolbox.prototype.init;
+
+/**
+ * Fill the toolbox with categories and blocks.
+ * @param {Array.<Blockly.utils.toolbox.Toolbox>} toolboxDef Array holding objects
+ *    containing information on the contents of the toolbox.
+ */
+Blockly.IToolbox.prototype.render;
+
+/**
+ * Dispose of this toolbox.
+ */
+Blockly.IToolbox.prototype.dispose;
+
+/**
+ * Get the width of the toolbox.
+ * @return {number} The width of the toolbox.
+ */
+Blockly.IToolbox.prototype.getWidth;
+
+/**
+ * Get the height of the toolbox.
+ * @return {number} The width of the toolbox.
+ */
+Blockly.IToolbox.prototype.getHeight;
+
+/**
+ * Get the toolbox flyout.
+ * @return {Blockly.Flyout} The toolbox flyout.
+ */
+Blockly.IToolbox.prototype.getFlyout;
+
+/**
+ * Move the toolbox to the edge.
+ */
+Blockly.IToolbox.prototype.position;
+
+/**
+ * Unhighlight any previously specified option.
+ * @return {void}
+ */
+Blockly.IToolbox.prototype.clearSelection;
+
+/**
+ * Updates the category colours and background colour of selected categories.
+ */
+Blockly.IToolbox.prototype.updateColourFromTheme;
+
+/**
+ * Adds a style on the toolbox. Usually used to change the cursor.
+ * @param {string} style The name of the class to add.
+ */
+Blockly.IToolbox.prototype.addStyle;
+
+/**
+ * Removes a style from the toolbox. Usually used to change the cursor.
+ * @param {string} style The name of the class to remove.
+ */
+Blockly.IToolbox.prototype.removeStyle;
+
+/**
+ * Update the flyout's contents without closing it.  Should be used in response
+ * to a change in one of the dynamic categories, such as variables or
+ * procedures.
+ */
+Blockly.IToolbox.prototype.refreshSelection;
+
+/**
+ * Toggles the visibility of the toolbox.
+ * @param {boolean} isVisible True if the toolbox should be visible.
+ */
+Blockly.IToolbox.prototype.setVisible;
```

Comment:
toolbox.js doesn't implement this method?

Reply:
good catch, must have stashed when i changed branches.